### PR TITLE
fix synchronizer unittest

### LINKED ISSUE #3018: adapt test for synchronizer: unittest and e2e-group 6 to etrog
# Rationale

The tests need to be adapted to the etrog changes

### PR DESCRIPTION
**Pull request by joanestebanr**
Closes #3018

### What does this PR do?
fix unittest

### Reviewers

Main reviewers:

- @ARR552 
- @tclemos

### CODE REVIEW

File: synchronizer/synchronizer.go [hunk 1]
```diff
@@ -392,9 +392,9 @@ func (s *ClientSynchronizer) Sync() error {
 						}
 					} else if err != nil && errors.Is(err, syncinterfaces.ErrMissingSyncFromL1) {
 						log.Info("Syncing from trusted node need data from L1")
-					} else {
+					} else if err != nil {
```
Comment (ARR552) [inline synchronizer/synchronizer.go]: Why do you move lines 382 and 383 inside this if?

File: synchronizer/actions/etrog/processor_l1_sequence_batches.go [hunk 2]
```diff
@@ -171,6 +171,19 @@ func (g *ProcessorL1SequenceBatchesEtrog) processSequenceBatches(ctx context.Con
...
+			batchL2Data := sbatch.PolygonRollupBaseEtrogBatchData.Transactions
+			processCtx = state.ProcessingContextV2{
+				BatchNumber: sbatch.BatchNumber,
+				Coinbase:    sbatch.SequencerAddr,
+				Timestamp:   &tstampLimit,
+				L1InfoRoot:  sbatch.PolygonRollupBaseEtrogBatchData.ForcedGlobalExitRoot,
+				//L1InfoTreeData:       leafs,
+				ForcedBatchNum:       batch.ForcedBatchNum,
+				BatchL2Data:          &batchL2Data,
+				SkipVerifyL1InfoRoot: 0,
```
Comment (ARR552) [inline synchronizer/actions/etrog/processor_l1_sequence_batches.go]: this must be 1

File: synchronizer/actions/etrog/processor_l1_sequence_batches.go [hunk 3]
```diff
@@ -171,6 +171,19 @@ func (g *ProcessorL1SequenceBatchesEtrog) processSequenceBatches(ctx context.Con
 				},
 				L1InfoTreeRoot: sbatch.PolygonRollupBaseEtrogBatchData.ForcedGlobalExitRoot,
 			}
+			//TODO: Check what we have to pass to executor for forcedBatch >1
+			tstampLimit := time.Unix(int64(sbatch.PolygonRollupBaseEtrogBatchData.ForcedTimestamp), 0)
+			batchL2Data := sbatch.PolygonRollupBaseEtrogBatchData.Transactions
```
Comment (ARR552) [inline synchronizer/actions/etrog/processor_l1_sequence_batches.go]: This should be the value stored here `forcedBatches`

File: synchronizer/actions/etrog/processor_l1_sequence_batches.go [hunk 4]
```diff
@@ -171,6 +171,19 @@ func (g *ProcessorL1SequenceBatchesEtrog) processSequenceBatches(ctx context.Con
 				},
 				L1InfoTreeRoot: sbatch.PolygonRollupBaseEtrogBatchData.ForcedGlobalExitRoot,
 			}
+			//TODO: Check what we have to pass to executor for forcedBatch >1
+			tstampLimit := time.Unix(int64(sbatch.PolygonRollupBaseEtrogBatchData.ForcedTimestamp), 0)
```
Comment (ARR552) [inline synchronizer/actions/etrog/processor_l1_sequence_batches.go]: this timestamp should be the one stored here `forcedBatches`

File: synchronizer/actions/etrog/processor_l1_sequence_batches.go [hunk 5]
```diff
@@ -171,6 +171,19 @@ func (g *ProcessorL1SequenceBatchesEtrog) processSequenceBatches(ctx context.Con
...
+			batchL2Data := sbatch.PolygonRollupBaseEtrogBatchData.Transactions
+			processCtx = state.ProcessingContextV2{
+				BatchNumber: sbatch.BatchNumber,
+				Coinbase:    sbatch.SequencerAddr,
+				Timestamp:   &tstampLimit,
+				L1InfoRoot:  sbatch.PolygonRollupBaseEtrogBatchData.ForcedGlobalExitRoot,
+				//L1InfoTreeData:       leafs,
```
Comment (ARR552) [inline synchronizer/actions/etrog/processor_l1_sequence_batches.go]: I'm checking with protocol if L1InfoData can be empty in this case

Comment (ARR552) [reply synchronizer/actions/etrog/processor_l1_sequence_batches.go]: I confirm this is correct. Leaves are no needed in forcedBatches

File: synchronizer/synchronizer.go (hunk 1 above)
Comment (joanestebanr) [reply synchronizer/synchronizer.go]: I refactor this if, but the code in the else only must be executed is the error is not `ErrFatalDesyncFromL1` or `ErrMissingSyncFromL1`

**Review by agnusmor — APPROVED** [review]

**Review by ARR552 — APPROVED** [review]

**Pull request merged**